runtime/compiler/aarch64/codegen/J9ARM64Snippet.cpp

```diff
@@ -74,7 +74,7 @@ TR::ARM64MonitorEnterSnippet::emitSnippetBody()
    *(int32_t *)buffer = TR::InstOpCode::getOpCodeBinaryEncoding(TR::InstOpCode::andimmx);
    tempReg->setRegisterFieldRD((uint32_t *)buffer);
    dataReg->setRegisterFieldRN((uint32_t *)buffer);
-   *(int32_t *)buffer |= 0x196000; // immr=25, imms=24 for 0xFFFFFF80
+   *(int32_t *)buffer |= 0x79E000; // immr=57, imms=56 for 0xFFFFFFFFFFFFFF80
```

Comment:
ppc uses `~OBJECT_HEADER_LOCK_RECURSION_MASK` which is `~0xF8` for masking.

https://github.com/eclipse/openj9/blob/944f9dd6edf869c6409db6ba367a360cad0dbee7/runtime/compiler/p/codegen/J9TreeEvaluator.cpp#L5925-L5948

Reply:
Thank you for pointing that out.
The monitorExit side has to use `~OBJECT_HEADER_LOCK_RECURSION_MASK` (= `~0xF8`) while the monitorEnter side uses `~0x7F`.

Updated the code.

Reply:
In monitorEnter, 32-bit ARM uses `~(OBJECT_HEADER_LOCK_BITS_MASK - OBJECT_HEADER_LOCK_LAST_RECURSION_BIT)`, which is `~0x7F`.
Power uses `LOCK_THREAD_PTR_AND_UPPER_COUNT_BIT_MASK` there, which is defined as `(LOCK_THREAD_PTR_MASK | OBJECT_HEADER_LOCK_LAST_RECURSION_BIT)` = `~0xFF | 0x80`.
It is confusing, but both resulting in the same mask value.